### DISCUSSION TITLE
Understanding of the concept Instance type

### DISCUSSION THREAD

**Discussion by suncpp** [json-schema-org/community#265] [Q&A] [answered]
### concept

1. **Instance Data Model**
>  JSON Schema interprets documents according to a data model. A JSON value interpreted according to this data model is called an "instance".
2. **Instance**
> A JSON document to which a schema is applied is known as an "instance".
3. **Instance Type**
> Numeric, Strings,	Arrays,Objects
4. **primitive type**
> null, boolean, object, array, number, 	string
--------------------------
### According to document
https://json-schema.org/draft/2020-12/json-schema-core.html
https://json-schema.org/draft/2020-12/json-schema-validation.html
----------------------------
### 问题
1. Instance inherit Instance Data Model, primitive type inherit Instance Type, Is it correct？
3. What is an Instance Type? How is it defined?
----------------------------
Look forward to your reply, thank you

**Comment by handrews** [comment, marked as answer]
@suncpp "instance type" is an informal term that means the primitive type of the current part of the instance. In the instance:

```
{
    "foo": 3,
    "bar": "hello"
}
```

* The instance type of the whole instance is object
* The instance type of "foo" is number
* The instance type of "bar" is string

In the validation specification, some keywords only apply to certain instance types, and are grouped into sections accordingly.

**Reply by suncpp** [reply]
Thank you for your patient reply
